src/main/kotlin/org/onflow/flow/sdk/models.kt

```diff
@@ -82,7 +82,9 @@ enum class HashAlgorithm(
     SHA2_256("SHA-256", 256, "SHA256withECDSA", 1, 1),
```

Comment:
Thinking more about my comment above, I'm not sure if my suggestions would be breaking changes to whoever is importing the repo. I haven't worked with this repo before and I'm not sure how it's being used, happy to hear your thoughts @lealobanov.

Reply:
I can check with one of our partners about the integration that they have

Reply:
I think we could get away with removing `code` if we update the protobuf schema to use a string and pass `algorithm` instead. I'm not sure if that may potentially introduce breaking changes if other repos are also relying on those protobuf class definitions...

Reply:
https://github.com/onflow/flow/blob/master/protobuf/flow/entities/account.proto - here under the definition for AccountKey the sign and hash algorithms are passed as int32

Reply:
I don't think `index` actually gets used anywhere in the implementation apart from being a zero-based counter (not sure if that is even needed, since the algorithm can be uniquely referred to by .algorithm field or the enum class definition itself). So we could potentially eliminate it assuming we wouldn't introduce breaking changes for existing key users.

Reply:
Will also update tests re. your comment above - thanks!

Reply:
- good point with `code` being used by protobuf in the `AccountKey` encoding. I think `index` could have been used for protobuf instead because index 0 is always assigned to the `unknown` algorithm which is never used (but it's too late to switch now because that may introduce a breaking change). Another option i to not use a field `code` but make that function `setHashAlgo` (on the screenshot) set a protofub integer by looking at the algorithm string.

- `index` seems to be used by [this function](https://github.com/onflow/flow-jvm-sdk/blob/main/src/main/kotlin/org/onflow/flow/sdk/models.kt#L69). It represents the Cadence code of algorithms (here are the [Cadence](https://cadence-lang.org/docs/language/crypto#hashing) codes), so I think it's better to keep the `index` field. 

Too bad the field names aren't self explanatory like `protobufCode` and `CadenceCode`. Maybe we could add a comment in the class definition to explain what the field is used for (at least what we found out so far)

Reply:
Yep, I also noticed that when I disabled sha2-384 and sha3-384 some of the existing tests for AccountKey were affected as the index/code shifted. Will play around with it a bit more tomorrow to see if I can get it working. Otherwise I agree that long term we could probably update protobuf "algorithm" as a string field for the unique identifier instead of code. For now will add some comments to the class definition as well.

Reply:
Could you please keep track of any and all changes that could be breaking in a file (crescendo-breaking-changes.md) at the top level?